org-code-javabuilder/lib/src/main/java/org/code/neighborhood/GridFactory.java

```diff
@@ -0,0 +1,77 @@
+package org.code.neighborhood;
+
+import java.io.File;
+import java.io.FileInputStream;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+public class GridFactory {
+  private static final String GRID_FILE_NAME = "grid.txt";
+  private static final String GRID_SQUARE_TYPE_FIELD = "tileType";
+  private static final String GRID_SQUARE_VALUE_FIELD = "value";
+
+  protected Grid createGridFromJSON(String filename) {
+    File file = new File(GRID_FILE_NAME);
+    FileInputStream fis;
+    try {
+      fis = new FileInputStream(file);
+      byte[] data = new byte[(int) file.length()];
+      fis.read(data);
+      fis.close();
+      return createGridFromString(new String(data, "UTF-8"));
+    } catch (Exception e) {
+      throw new UnsupportedOperationException("Try adding a grid.txt file.");
```

Comment:
Is this the only reason we'd hit an exception?

Reply:
oh you're totally right, we could hit exceptions from createGridFromString and we shouldn't catch those here. The only one we should be catching is from reading the file.